### PR TITLE
Mongodb/configure read concern

### LINKED ISSUE #17805: Make it possible to configure read concern for mongodb database/ client
## Description

Hello there,

similar to the write concern, which is configurable via the application properties, it would be really great to be able to configure the read concern for mongodb databases.

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #17805

### CODE REVIEW

**Review by evanchooly — APPROVED** [review]
a couple of questions I think i already know the answer to but otherwise looks fine.

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientConfig.java [hunk 1]
```diff
@@ -171,6 +171,13 @@
     @ConfigDocSection
     public WriteConcernConfig writeConcern;
 
+    /**
+     * Configures the read concern.
+     * Supported values are: {@code local|majority|linearizable|snapshot|available}
+     */
+    @ConfigItem
+    public Optional<String> readConcern;
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientConfig.java:179]: What's the default in mongo?

**Review by cescoffier — APPROVED** [review]

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (loicmathieu) [reply extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientConfig.java:179]: A read concern is built from the ReadConcernLevel enum.
There is a DEFAULT read concern, but no default level ...

I'm not a big fan of the optional usage but it's aligned with what we did for readPreference and the fact that there is a mismatch between the ReadConcern and the ReadConcernLevel makes it difficult to use the later.

**Comment by evanchooly** [comment]
It looks like for reads against the primary (the usual/recommended case), the default is `local`.  

https://docs.mongodb.com/manual/reference/mongodb-defaults/

**Comment by cescoffier** [comment]
With optional (not a fan either), we can detect when it's not set, and let Mongo decide the default. So, if one day, Mongo changes the default, we are good (ok, it's unlikely, but well).

**Comment by loicmathieu** [comment]
@evanchooly if we use `local` as our default it will not be the same as the default value which depends on the read preference, if read preference is on secondary the default read concern will be `available`.
So I think the best is to keep the optional.

**Pull request merged**